### PR TITLE
IW-1118 Trim whitespace from selections when creating hyperlinks

### PR DESCRIPTION
**Pull request by jsutterfield**
Ticket: https://wikia-inc.atlassian.net/browse/IW-1118

Trim leading and trailing whitespace from the selection when creating hyperlinks. See [this comment](https://wikia-inc.atlassian.net/browse/IW-1118?focusedCommentId=439715&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-439715) for more details.

cc @Wikia/iwing

### CODE REVIEW

File: src/components/HyperlinkingWrapper.js [hunk 1]
```diff
@@ -193,6 +197,17 @@ export default class HyperlinkingWrapper extends Component {
         this.onClose();
     }
 
+    getRangeMinusWhitespace(range) {
+        const selectedText = this.quill.getText(range);
+        const leadingWhitespaceCount = selectedText.length - selectedText.trimLeft().length;
```
Comment (xkxd) [inline src/components/HyperlinkingWrapper.js]: `trimLeft/Right` will blow on IE without polyfill https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimStart

Comment (latata) [reply src/components/HyperlinkingWrapper.js]: I'd go with regexp solution 
```suggestion
        const leadingWhitespaceCount = selectedText.length - selectedText.replace(/^\s*/, '').length;
```

File: src/components/HyperlinkingWrapper.js [hunk 2]
```diff
@@ -193,6 +197,17 @@ export default class HyperlinkingWrapper extends Component {
...
+        const trailingWhitespaceCount = selectedText.length - selectedText.trimRight().length;
```
Comment (latata) [inline src/components/HyperlinkingWrapper.js]:
```suggestion
        const trailingWhitespaceCount = selectedText.length - selectedText.replace(/\s*$/, '').length;
```

**Comment by jsutterfield** [comment]
Whoops, thanks for the rebuild @xkxd

**Review by latata — APPROVED** [review]

**Comment by latata** [comment]
Closing at it's been fixed on backend side.

**Pull request closed without merging**